src/core/context/useChat.tsx

```diff
@@ -23,16 +23,13 @@ import type { ChatCompletionMessageParam } from 'openai/resources/index';
 
 let client: OpenAI | null = null;
 
-export const AppContextProvider = (props: {
-  children: React.ReactNode;
-  thinkingStore: TextStreamStore;
-  errorStore: TextStreamStore;
-  messageStore: TextStreamStore;
-  progressStore: TextStreamStore;
-  messageHistoryStore: MessageHistoryStore;
+export const useChat = (props?: {
+  thinkingStore?: TextStreamStore;
+  errorStore?: TextStreamStore;
+  messageStore?: TextStreamStore;
+  progressStore?: TextStreamStore;
+  messageHistoryStore?: MessageHistoryStore;
```

Comment:
Interesting to see the difference between this and Hard AI implementation. Maybe after some additional refactoring they will become more similar.